src/resmom/linux/alps.c

```diff
@@ -3650,31 +3650,33 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 					/* Only count them if the state is UP */
 					totaccel++;
 			}
+
 			attr = "resources_available.naccelerators";
-			if (totseg == 0) {
+			if (totseg == 0 || vnode_per_numa_node) {
 				/*
 				 * add the naccelerators count only to
 				 * the first vnode of a compute node
 				 * all other vnodes will share the count
 				 */
 				snprintf(utilBuffer, sizeof(utilBuffer),
 					"%d", totaccel);
-			} else if (vnode_per_numa_node) {
-				/*
-				 * When there is a vnode being created
-				 * per numa node, only the first
-				 * (segment 0) vnode gets the accelerator.
-				 * The other vnodes must
-				 * share the accelerator count with
-				 * segment 0 vnodes
-				 */
-				snprintf(utilBuffer, sizeof(utilBuffer),
-					"@%s_%ld_0",
-					mpphost, node->node_id);
+				if (vnode_per_numa_node && !totseg) {
+					/*
+					 * When there is a vnode being created
+					 * per numa node, only the first
+					 * (segment 0) vnode gets the accelerator.
+					 * The other vnodes must
+					 * share the accelerator count with
+					 * segment 0 vnodes
+					 */
+					snprintf(utilBuffer, sizeof(utilBuffer),
+						 "@%s_%ld_0",
+						 mpphost, node->node_id);
+				}
+				if (vn_addvnr(nv, vname, attr, utilBuffer,
+					      0, 0, NULL) == -1)
+					goto bad_vnl;
```

Comment:
I don't understand why this code got changed.  The code also seems to produce the wrong result.  When creating NUMA node 0's vnode, you will set the naccelerator to "@mpphost_nodeid_0".  In other words you say it shares with itself.  And all the other NUMA nodes for the node will get an actual value.  This does not match the existing behavior of the OSS code.